resolwe/storage/manager.py

```diff
@@ -28,6 +28,15 @@ def _lock_file_storage(self, file_storage_id: int) -> Optional[FileStorage]:
 
     def delete_single_location(self, file_storage: FileStorage, connector_name: str):
         """Delete given storage location."""
+        # Some other storage manager might have already processed this storage
+        # location before we did.
+        if not file_storage.storage_locations.filter(
+            connector_name=connector_name,
+            url=file_storage.subpath,
+        ).exists():
```

Comment:
This is still not atomic. You should do it like:
```py
delete_location = <queryset>.first()
if not delete_location:
    return
```

Reply:
Are you sure about that? 

The FileStorage object is locked so no other manager can process it at the same time.

Reply:
You are right. But it is still nicer to make fewer queries.